src/Nethermind/Nethermind.Evm/Tracing/GethStyle/Custom/Native/GethLikeNativeTracerFactory.cs

```diff
@@ -3,18 +3,21 @@
 
 using System;
 using System.Collections.Generic;
-using Nethermind.Evm.Tracing.GethStyle.Custom.Native.Tracers;
+using Nethermind.Evm.Tracing.GethStyle.Custom.Native.FourByte;
+using Nethermind.Evm.Tracing.GethStyle.Custom.Native.Prestate;
+using Nethermind.State;
+
 namespace Nethermind.Evm.Tracing.GethStyle.Custom.Native;
 
 public static class GethLikeNativeTracerFactory
 {
     static GethLikeNativeTracerFactory() => RegisterNativeTracers();
 
-    private static readonly Dictionary<string, Func<GethTraceOptions, GethLikeNativeTxTracer>> _tracers = new();
+    private static readonly Dictionary<string, Func<IWorldState?, NativeTracerContext?, GethTraceOptions, GethLikeNativeTxTracer>> _tracers = new();
 
-    public static GethLikeNativeTxTracer CreateTracer(GethTraceOptions options) =>
-        _tracers.TryGetValue(options.Tracer, out Func<GethTraceOptions, GethLikeNativeTxTracer> tracerFunc)
-        ? tracerFunc(options)
+    public static GethLikeNativeTxTracer CreateTracer(IWorldState? worldState, NativeTracerContext? context, GethTraceOptions options) =>
```

Comment:
would not make them nullable here

Reply:
Since the 4byteTracer doesn't need worldState or context I decided to not to pass them in when not needed. However, to use to be able to register the tracers from `GethLikeNativeTracerFactory` I decided to just pass them in as `_` in the delegate for when worldState or context aren't required and don't need to be passed through